linkerd/detect/src/lib.rs

```diff
@@ -116,15 +133,18 @@ where
     }
 
     fn call(&mut self, mut io: I) -> Self::Future {
-        let mut inner = self.inner.clone();
-        let mut buf = BytesMut::with_capacity(self.capacity);
-        let detect = self.detect.clone();
+        let Config {
+            detect,
+            capacity,
+            timeout,
+        } = self.config.clone();
         let target = self.target.clone();
-        let timeout = self.timeout;
+        let mut inner = self.inner.clone();
         Box::pin(async move {
             trace!("Starting protocol detection");
```

Comment:
nit, take it or leave it: now that the config parameters can be dynamic, maybe this event should include them:
```suggestion
            trace!(
                config.capacity = capacity,
                config.timeout = ?timeout,
                "Starting protocol detection"
            );
```